zerver/views/upload.py

```diff
@@ -71,16 +58,29 @@ def serve_file_backend(request, user_profile, realm_id_str, filename, redir):
 
     return serve_s3(request, user_profile, realm_id_str, filename, redir)
 
-@authenticated_json_post_view
-@has_request_variables
-def json_upload_file(request, user_profile):
-    # type: (HttpRequest, UserProfile) -> HttpResponse
-    return upload_file_backend(request, user_profile)
-
 @zulip_login_required
-@has_request_variables
```

Comment:
"request variables" are those special REQ arguments.  `upload_file_backend` doesn't actually have any, so it's not needed.